flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -1060,21 +1061,73 @@ private static Function instantiateHiveFunction(ObjectPath functionPath, HiveCat
 		);
 	}
 
+	private boolean isTablePartitioned(Table hiveTable) {
+		return hiveTable.getPartitionKeysSize() != 0;
+	}
+
 	// ------ stats ------
 
 	@Override
 	public void alterTableStatistics(ObjectPath tablePath, CatalogTableStatistics tableStatistics, boolean ignoreIfNotExists) throws TableNotExistException, CatalogException {
-
+		try {
+			Table hiveTable = getHiveTable(tablePath);
+			// Set table stats
+			if (needToUpdateStatistics(tableStatistics, hiveTable.getParameters())) {
+				updateStatisticsParameters(tableStatistics, hiveTable.getParameters());
+				client.alter_table(tablePath.getDatabaseName(), tablePath.getObjectName(), hiveTable);
```

Comment:
Have you tested this with Hive-1.2.1? I had some issues when I implemented this for 1.2.1 internally, and added a shim for it.

Reply:
No, i haven't tested this with Hive-1.2.1, since you have added a shim for it, I'll add it also.